### PR TITLE
Improve step5 (docs and frontend)

### PR DESCRIPTION
**Pull request by tkat0**
## What

- `document/step5.{en,ja}.md`
  - Fix typo and add some notes for React
  - Change recommended node version from v14 to v16
    - To avoid environmental troubles as node v14 and later sometimes behave differently
- `typescript/*`
  - Change to reload after listing complete
    - I guess the behavior before the change confuses trainees. I thought having the trainees fix it but React is out of scope in this training. So I fixed it.

## CHECKS :warning:

Please make sure you are aware of the following.

- [x] **The changes in this PR doesn't have private information

### CODE REVIEW

File: document/step5.en.md [hunk 1]
```diff
@@ -41,8 +43,8 @@ In this screen, item images are all rendered as Build@Mercari logo. Specify the
 
 
 ### (Optional) Task 3. Change the styling with HTML and CSS
-These two components are styled by CSS. To see what types of changes can be made, try modifying `ItemList` component CSS. These are specifed in `App.css` and they are applied by `className` attribute (e.g. `<div>className='Listing'</div>`).
-```
+These two components are styled by CSS. To see what types of changes can be made, try modifying `ItemList` component CSS. These are specifed in `App.css` and they are applied by `className` attribute (e.g. `<div className='Listing'></div>`).
```
Comment (KengoA) [inline document/step5.en.md:46]: good catch 🙇

File: typescript/simple-mercari-web/src/App.tsx [hunk 2]
```diff
@@ -1,9 +1,11 @@
-import React, { useEffect, useMemo, useState } from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { ItemList } from './components/ItemList';
 import { Listing } from './components/Listing';
 
 function App() { 
+  // reload ItemList after Listing complete
+  const [reload, setReload] = useState(true);
```
Comment (KengoA) [inline typescript/simple-mercari-web/src/App.tsx:8]: thank you for adding reload I missed this

**Review by KengoA — APPROVED** [review]
LGTM!

**Comment by tkat0** [comment]
Thank you for your review!

**Pull request merged**